robosuite/eval/eval.py

```diff
@@ -0,0 +1,114 @@
+import os
+import h5py
+import mediapy as media
+import matplotlib.pyplot as plt
+import numpy as np
+import random
+import os
+import openai
+
+# openai.api_key = os.getenv("OPENAI_API_KEY")
+
+
+def load_hdf5(dataset_path: str) -> tuple:
+    if not os.path.isfile(dataset_path):
+        print(f"Dataset does not exist at \n{dataset_path}\n")
+        exit()
+
+    with h5py.File(dataset_path, "r") as root:
+        qpos = root["/observations/qpos"][()]
+        qvel = root["/observations/qvel"][()]
+        image_dict = dict()
+        for cam_name in root[f"/observations/images/"].keys():
+            image_dict[cam_name] = root[f"/observations/images/{cam_name}"][()]
+        action = root["/action"][()]
+
+    return qpos, qvel, action, image_dict
+
+
+def find_stopping_point(data: np.ndarray) -> int:
+    steps, _ = data.shape
+    for step in range(steps):
+        if np.all(data[step:, :] == 0):
+            return step
+
+    return data.shape[0]
+
+
+def sample_frames(qpos: np.ndarray, action: np.ndarray, num_samples: int) -> list:
+    arr = qpos[:, 7]
+    num_choices = num_samples // 3 - 1
+
```

Comment:
Feel free to add later but, for edge cases where the gripper does not open,(to avoid panics) return a random sampling.